CONTRIBUTING.md

```diff
@@ -0,0 +1,209 @@
+
+### Running tests
+
+The tests are divided into `unit` tests, whose aim is to check individual bits of code, and end-to-end tests, which check how parts of the program interact as a whole.
+
+To run all tests,
+```bash
+nosetests beep
+```
+To run a specific test script
+```bash
+nosetests test_featurize.py
+```
+To run a specific test from a test class,
+``` bash
+nosetests test_structure.py:TestFeaturizer.test_insufficient_data_file
+```
+If you want to run several, but not all, the tests from a script, you can restrict which tests are run from a particular script by using the skipping decorator:
+```python
+@unittest.skip("")
+def test_bit_of_code(self):
+    ...
+```
+or by just commenting out all the tests you don't want to run
+
+### Debugging tests
+
+-  Set break points, either in your IDE or using the python debugging module. To use the latter, add the following line where you want to set the break point This will start the [Python interactive debugger](https://gist.github.com/mono0926/6326015).
+```python
+import ipdb; ipdb.set_trace()
+```
+- Warnings: If functions are raising warnings instead of errors, it can be hard to pinpoint where this is coming from. Here, you can use the `warnings` module to convert warnings to errors:
+  ```python
+  import warnings
+  warnings.simplefilter("error")
+  ```
+  Then you can use a try-except block, as in a., but with, for example, `RuntimeWarning` instead of `ValueError`.
+
+### Profiling
+
+
+## Documentation
+
+### Building the documentation
+
+
+### Tutorial notebooks
+
+Specific use-cases and capabilities of BEEP are showcased in Jupyter notebooks stored in the [examples directory](beep/tutorials). 
+All example notebooks should be listed in [tutorials/README.md](https://github.com/TRI-AMDD/beep/tree/master/beep/tutorials). 
+
+
+## Infrastructure
+
+### Setuptools
+
+Installation _and dependencies_ are handled via [setuptools](http://setuptools.readthedocs.io/)
+
+Configuration files:
+
+```
+setup.py
+```
+
+Note that this file must be kept in sync with the version number in [beep/__init__.py](beep/__init__.py).
+
+### Continuous integration
+
+All committed code is tested using [Travis CI](https://travis-ci.org/), tests are published [here](https://travis-ci.com/github/TRI-AMDD/beep/).
+Configuration file: ``` .travis.yaml```. For every commit, Travis runs all the unit tests, end to end tests, doc tests and flake8.
+<!-- Unit tests and flake8 testing is done for every commit. A nightly cronjob also tests the notebooks. Notebooks listed in `.slow-books` are excluded from these tests. -->
+
+Additionally, Appveyor runs integration tests for windows environment. Tests are published [here](https://ci.appveyor.com/project/TRI-AMDD/beep)
+
+
+### Codecov
```

Comment:
This isn't quite right for our setup (we don't have a .coveragerc, but we could make it).

Reply:
Fixed.